conclave-cloud-auction/README.md

```diff
@@ -0,0 +1,285 @@
+The developer writes the code for the user. The developer generates a hash value for this code, the user inspects the code either himself or with the help of some auditor, and makes sure the code does what its expected to do.The user notes down this hash.
+
+#### Step 2:
+The developer requests a public key and attestation report of the KDS from the KDS .
+The developer encrypts the function which is to be uploaded using this public key and uploads it to the conclave cloud platform. This function will sit in a database in the cloud.
+
+#### Step 3:
+The user can now invoke this function specifying the expected hash of the function., this is when the function is provisioned into an action enclave, ie an enclave is initialized , started with this function.
+The platform will encrypt the invocation input parameters and the hash of the function using the same public key received by the KDS.
+
+#### Step 4:
+The action enclave requests the private key from the KDS. The KDS first validates the attestation report from enclave. Enclave also validates Kds s attestation report. Both mutually remote attest each other. KDS also releases the private key to the enclave if it matches the policy constarints.
+The action enclave will decrypt the function, expected function hash and parameters using this private key inside the enclave. Platform provisions this decrypted function inside the JS engine.
+The platform verifies if the hash of the action enclave matches with that provided by the user. If they match then only the function is executed and response sent back to the client. It encrypts the response to be sent to the client using clients public key.
+
+Thats how end to end encryption works.
+
+
+## Building and deploying the demonstration
+Each component must be built and deployed or hosted in order to run the
+demonstration.
+
+### 1. Backend to allow Conclave Functions to access it for storing user databases.
+
+It is recommended to set up a new virtual machine with your cloud service
+provider to host this. The resource requirements are tiny so you can use the
+smallest size of virtual machine your cloud service provider provides.
+
+The service is built using this command:
+
+```
+./gradlew build
+```
+
+You can then deploy the `build/libs/conclaveauction-0.0.1-SNAPSHOT.jar ` file to
+your virtual machine and run it with:
+
+```
+java -jar ./conclaveauction-0.0.1-SNAPSHOT.jar 
+```
+
+### 2. Functions
+Note down the IP address for the backend service that you have just deployed and
+update the functions module to connect to the service with that address in
+`functions/src/index.ts`.
+
+The package should be cloned into a new directory, then the dependencies
+installed with:
+
+```
+npm install
+```
+
+The package can then be built using:
+
+```
+npm run build
+```
+##### Uploading the functions
+Once the package has been built it needs to be uploaded to a Conclave Cloud
+project. This is achieved using the Conclave Cloud CLI.
```

Comment:
Maybe have a link to the Conclave Cloud CLI on the confluence page. The link can be changed later when the CLI is published somewhere else.